### ISSUE TITLE
additional testing for zk detection

### ISSUE THREAD

**Issue by jdef** [mesos/mesos-go#100]
Was thinking about the implementation some more, and am concerned that the unit tests are still not adequate. for example:

* what if the client's embedded connection flaps between connected/disconnected - what does that look like to the detector (different from #99, which tests the whole connector flapping)?
* what about other failure modes of the embedded connection, should `client.watchChildren` react more intelligently?

**Comment by vladimirvivien** [comment]
@jdef with your integration of this with Kubernetes, do you still feel uncomfortable, should we revisit ?

**Comment by jdef** [comment]
christos had mentioned that he experienced some problems with the detector when ZK was killed and started. I tried with a single-master system (on GCE) and the mesos master died too - which I thought was strange. I haven't tested on a multi-ZK cluster yet but I wanted to start a framework and then take down all the ZK nodes, then bring them back up and see what happened. And from there, perhaps derive some additional automated tests.

**Comment by jdef** [comment]
additional testing:
* https://github.com/mesos/mesos-go/commit/049b57bd9bd0550056f26b5ced76f36a820243eb

**Comment by tsenart** [comment]
@jdef: Is this covered in #162?

**Comment by jdef** [comment]
No, but #162 may be seen as an enabler because it's MUCH thinner than the
fat client shim that was there before and the interface is cleaner.

**Comment by tsenart** [comment]
This is another candidate issue to be moved to an independent `github.com/mesosphere/detector`
